plugins/plugin-utils/src/executors-utils.ts

```diff
@@ -138,10 +137,15 @@ export function batchWorkerExecutor<ExecutorOptions>(workerModule: string) {
 
         // Run yield loop after dispatch to avoid serializing execution.
         for (let taskIndex = 0; taskIndex < tasks.length; taskIndex++) {
-            const taskName = tasks[taskIndex++];
+            const taskName = tasks[taskIndex];
             yield results.get(taskName);
         }
 
+        await Promise.allSettled([...results.values()]);
```

Comment:
Minor but I think you can just do this

```suggestion
        await Promise.allSettled(results.values());
```